client/src/components/reader/Resource/Viewer/Wrapper.js

```diff
@@ -26,6 +27,7 @@ export default class ResourceViewerWrapper extends PureComponent {
     if (this.props.annotations !== nextProps.annotations) return true;
     if (this.props.activeAnnotation !== nextProps.activeAnnotation) return true;
     if (this.props.resources !== nextProps.resources) return true;
+    if (this.props.authentication !== nextProps.authentication) return true;
```

Comment:
See my comment above about the RequireRole component... generally, just flowing more props down the tree may not be the right solution.